packages/teams-js/src/internal/communication.ts

```diff
@@ -69,6 +80,7 @@ export function initializeCommunication(validMessageOrigins: string[] | undefine
     Communication.currentWindow.parent !== Communication.currentWindow.self
       ? Communication.currentWindow.parent
       : Communication.currentWindow.opener;
+  Communication.topWindow = Communication.currentWindow.top;
```

Comment:
One other thing I forgot to mention: in some cases there is *no* parent window to talk to (mobile scenarios for example). In that case, the current window has a handler on it for message passing to use instead of post messages

Reply:
As discussed offline, NAA is limited to windows with a DOM. So, as of this PR, frameless windows won't be considered.